### ISSUE TITLE
Tokens are not stored for Extended Provider

### ISSUE THREAD

**Issue by pura** [hybridauth/hybridauth#918]
**Bug:**
When I create a new Provider (I'm working with AzureAd at the moment), the default storage expects the key to be name of a class. it does that by checking for 'Hybridauth\\Provider\\' in the current class namespace and replace that by empty string ''. 
But if the class namespace is different, then it keeps the whole namespace as providerId, which doesn't work as a session Key.

**Version and provider:**
3.0-RC2 , Custom Provider

**How can we replicate this issue?**
- Create a Custom provider with different namespace and use default session.
- The access_token is not stored

** Solution **
I can prepare a pull request as can replace (on AbstactAdapter)
```
$this->providerId = str_replace('Hybridauth\\Provider\\', '', get_class($this));
```
with something like (may be not exactly)
``
$this->providerId = !empty($this->providerId) ? $this->providerId: str_replace('Hybridauth\\Provider\\', '', get_class($this));

**Comment by ApacheEx** [comment]
Hi, 
thanks for reporting, 
I think you are right.

perhaps it's better to get `provderId` like this:
```php
$this->providerId = (new \ReflectionClass($this))->getShortName();
```

this will always return the class name of $this object and doesn't matter which namespace is.

**Comment by ApacheEx** [comment]
this fix will be available in 3.0 version. 
Please check if it works for you.
